db/migrate/20180626125655_rename_user_maintenance_zone_record.rb

```diff
@@ -0,0 +1,97 @@
+class RenameUserMaintenanceZoneRecord < ActiveRecord::Migration[5.0]
+  class UniqueWithinRegionValidator < ActiveModel::EachValidator
+    def validate_each(record, attribute, value)
+      return if value.nil?
+      match_case = options.key?(:match_case) ? options[:match_case] : true
+      record_base_class = record.class.base_class
+      field_matches =
+        if match_case
+          record_base_class.where(attribute => value)
+        else
+          record_base_class.where(record_base_class.arel_attribute(attribute).lower.eq(value.downcase))
+        end
+      unless field_matches.in_region(record.region_id).where.not(:id => record.id).empty?
+        record.errors.add(attribute, "is not unique within region #{record.region_id}")
+      end
+    end
+  end
+
+  class Zone < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    validates_presence_of :name
+    validates :name, "rename_user_maintenance_zone_record/unique_within_region" => true
+
+    MAINTENANCE_ZONE_NAME = "__maintenance__".freeze
+  end
+
+  class Job < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :miq_task, :dependent => :delete
+    after_update_commit :update_linked_task
+
+    def update_linked_task
+      miq_task&.update_attributes!(:zone => zone)
+    end
+  end
+
+  class MiqTask < ActiveRecord::Base
+    has_one :job, :dependent => :destroy
+    has_one :miq_queue
+  end
+
+  class MiqQueue < ActiveRecord::Base
+    belongs_to :miq_task
+  end
+
+  def up
+    say_with_time("Renaming user-defined Maintenance Zone") do
+      zone = Zone.in_my_region.find_by(:name => Zone::MAINTENANCE_ZONE_NAME)
+      if zone.present?
+        # STEP 1 - rename zone
```

Comment:
I still think this is the wrong approach

Reply:
@bdunne you were worried about string associations
1) possibility that user has `__maintenance__` as name is almost zero
2) if so, zone and associations are renamed to name with possibility of duplication zero^2.

could you be more specific? When this fails (with significant probability). 
Is it not acceptable for @Fryguy, @agrare, @carbonin too?

Reply:
> 1. possibility that user has __maintenance__ as name is almost zero

I agree, but while unlikely, not impossible

> 2. if so, zone and associations are renamed to name with possibility of duplication zero^2.

Right now you're only updating `Job` and `MiqQueue`.  The following models have a `zone` column:
```
=> ["MiqQueue", "Job", "MiqTask", "ManageIQ::Providers::Kubernetes::ContainerManager::Scanning::Job", "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::PlaybookRunner", "ManageIQ::Providers::EmsRefreshWorkflow", "ManageIQ::Providers::AnsibleRunnerWorkflow", "VmScan", "ManageIQ::Providers::NativeOperationWorkflow", "Service::LinkingWorkflow"]
```
I also believe it's in Settings and who knows how many other places it's buried in a serialized column.  I just don't think it's practical to try renaming a zone